packages/babel-plugin-transform-destructuring/src/util.ts

```diff
@@ -0,0 +1,654 @@
+import { types as t } from "@babel/core";
+import type { File } from "@babel/core";
+import type { Scope, NodePath } from "@babel/traverse";
+import type { TraversalAncestors } from "@babel/types";
+
+/**
+ * Test if an ArrayPattern's elements contain any RestElements.
+ */
+
+function hasRest(pattern: t.ArrayPattern) {
+  for (const elem of pattern.elements) {
+    if (t.isRestElement(elem)) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
Nit: these could be "compacted" to `pattern.elements.some(elem => t.isRestElement(elem))`